statesync/syncer.go

```diff
@@ -139,17 +139,19 @@ func (s *syncer) RemovePeer(peer p2p.Peer) {
 }
 
 // SyncAny tries to sync any of the snapshots in the snapshot pool, waiting to discover further
-// snapshots if none were found and discoveryTime > 0. It returns the latest state and block commit
+// snapshots if none were found for 5 seconds. It returns the latest state and block commit
 // which the caller must use to bootstrap the node.
-func (s *syncer) SyncAny(discoveryTime time.Duration, retryHook func()) (sm.State, *types.Commit, error) {
-	if discoveryTime != 0 && discoveryTime < minimumDiscoveryTime {
-		discoveryTime = 5 * minimumDiscoveryTime
+//
+// If none snapshots are found after maxDiscoveryTime, errNoSnapshots is returned.
+func (s *syncer) SyncAny(maxDiscoveryTime time.Duration, retryHook func()) (sm.State, *types.Commit, error) {
+	if maxDiscoveryTime < discoveryTime {
+		s.logger.Info(fmt.Sprintf("max_discovery_time is less than %v, overwriting...", discoveryTime))
 	}
 
-	if discoveryTime > 0 {
-		s.logger.Info("Discovering snapshots", "discoverTime", discoveryTime)
-		time.Sleep(discoveryTime)
-	}
+	timeStart := time.Now()
+
+	s.logger.Info(fmt.Sprintf("Discovering snapshots for %v", discoveryTime))
+	time.Sleep(discoveryTime)
```

Comment:
We always sleep here. Before we didn't if `discoveryTime == 0`.

Reply:
> discoveryTime == 0

True. But the default value was 5sec. So, to skip time here, a user needed to change the variable's value in the config.